File: 1952 three-divisors/solution.go
func isThree(n int) bool {
    root := int(math.Sqrt(float64(n)))
    if root*root != n {
        return false
    }
    return isPrime(root, 2)
}

func isPrime(num, d int) bool {
    if num < 2 {
        return false
    }
    if d*d > num {
        return true
    }
    if num%d == 0 {
        return false
    }
    return isPrime(num, d+1)
}

File: 1952 three-divisors/solution.java
class Solution {
    public boolean isThree(int n) {
        int root = (int)Math.sqrt(n);
        if (root * root != n) return false; 
        return isPrime(root, 2);            
    }

    private boolean isPrime(int num, int divisor) {
        if (num < 2) return false;
        if (divisor * divisor > num) return true;
        if (num % divisor == 0) return false;
        return isPrime(num, divisor + 1);
    }
}

